mail_tracking/static/src/js/failed_message/widget.js

```diff
@@ -0,0 +1,311 @@
+/* Copyright 2019 Alexandre Díaz
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). */
+odoo.define('mail_tracking.FailedMessage', function (require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var BasicModel = require('web.BasicModel');
+    var BasicView = require('web.BasicView');
+    var Chatter = require('mail.Chatter');
+    var MailThread = require('mail.widget.Thread');
+    var utils = require('mail.utils');
+    var core = require('web.core');
+    var field_registry = require('web.field_registry');
+    var time = require('web.time');
+
+    var QWeb = core.qweb;
+
+
+    /**
+    * Helper method to fetch failed messages
+    *
+    * @private
+    * @param {Object} self
+    * @param {Array} ids
+    * @returns {Array}
+    */
+    function _readMessages (self, ids) {
+        if (!ids.length) {
+            return $.when([]);
+        }
+        var context = self.record && self.record.getContext();
+        return self._rpc({
+            model: 'mail.message',
+            method: 'get_failed_messages',
+            args: [ids],
+            context: context || self.getSession().user_context,
+        }).then(function (messages) {
+            // Convert date to moment
+            _.each(messages, function (msg) {
+                msg.date = moment(time.auto_str_to_date(msg.date));
+                msg.hour = utils.timeFromNow(msg.date);
+            });
+            return messages;
+        });
+    }
+
+    BasicModel.include({
+
+        /**
+        *  Fetches the failed messages displayed by the 'mail_failed_message'
+        * field widget in form views.
+        *
+        * @private
+        * @param {Object} record
+        * @param {String} fieldName
+        * @returns {Array}
+        */
+        _fetchSpecialFailedMessages: function (record, fieldName) {
+            var localID = record._changes && fieldName in record._changes
+                ? record._changes[fieldName] : record.data[fieldName];
+            return _readMessages(this, this.localData[localID].res_ids);
+        },
+    });
+
+    var AbstractFailedMessagesField = AbstractField.extend({
+
+        /**
+        * Mark failed message as reviewed. By default should be don't reviewed.
+        *
+        * @returns {Boolean}
+        */
+        _markFailedMessageReviewed: function () {
+            return false;
+        },
+    });
+
+    var FailedMessage = AbstractFailedMessagesField.extend({
+        className: 'o_mail_failed_message',
+        events: {
+            'click .o_failed_message_retry': '_onRetryFailedMessage',
+            'click .o_failed_message_reviewed': '_onMarkFailedMessageReviewed',
+        },
+        specialData: '_fetchSpecialFailedMessages',
+
+        /**
+         * Overrides to listen bus notifications
+         *
+         * @Override
+         */
+        init: function () {
```

Comment:
Shouldn't you start listening to updates after the widget has been rendered? :thinking: 

```suggestion
        start: function () {
```